src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -28,7 +28,7 @@ class Crystal::Doc::Generator
     },
   }
 
-  def initialize(@program : Program, @included_dirs : Array(String), @dir = "./doc")
+  def initialize(@program : Program, @included_dirs : Array(String), @dir = "./docs")
```

Comment:
Maybe don't provide a default here, since it's also in `command/docs.cr`

Reply:
Yeah this duplication is bad. But you'd also want the default value if the generator is directly used from code.
I was thinking about adding a dedicated struct for holding option values. This could be initialized (with implicit defaults) instead of `output_directory = "./docs"` and be used as the second argument for `Doc::Generator.new`.
```crystal
struct Crystal::Doc::Generator::Options
  property included_dirs = [] of String
  property output_dir = "./docs"
end

Reply:
Or set the default as `nil` and then do `@dir ||= "./docs"`. And in the option parser place initialize it to `nil`.

Reply:
This feels dirty :P I'm working on an options container... ;)

Reply:
Oh, I meant:

```crystal
def initialize(dir = nil)
  @dir = dir || "./docs"
end
```

I don't think that's dirty at all.

Or remove the default value. The doc generator isn't used outside the compiler, and if it is, passing a mandatory directory string should be fine.

Adding an extra type complicates the design.

Reply:
I really like including the default value in the help message: `Set the output directory (default: ./docs)` For this to work, we'll need the value outside of the Generator constructor. So no default value there.

I feel like a dedicated options type is a cleaner solution on the long run. But I'm not fixed on that.